test/t1.ml

```diff
@@ -6,12 +6,28 @@ let run () =
 
   for _i = 1 to 50 do
     Trace.with_span ~__FILE__ ~__LINE__ "outer.loop" @@ fun _sp ->
+    let pseudo_async_sp =
+      Trace.enter_manual_toplevel_span ~__FILE__ ~__LINE__ "fake_sleep"
+    in
+
     for _j = 2 to 5 do
       incr n;
       Trace.with_span ~__FILE__ ~__LINE__ "inner.loop" @@ fun _sp ->
       Trace.messagef (fun k -> k "hello %d %d" _i _j);
       Trace.message "world";
-      Trace.counter_int "n" !n
+      Trace.counter_int "n" !n;
+
+      if _j = 2 then (
+        (* fake micro sleep *)
+        let _sp =
+          Trace.enter_manual_sub_span ~parent:pseudo_async_sp ~__FILE__
+            ~__LINE__ "sub-sleep"
+        in
+        Thread.delay 0.005;
+        Trace.exit_manual_span _sp
+      ) else if _j = 3 then
+        (* pretend some task finished. Note that this is not well scoped wrt other spans. *)
+        Trace.exit_manual_span pseudo_async_sp
```

Comment:
👏 I think this turned out really well, tbh

Reply:
:)